application/clicommands/DownloadCommand.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+namespace Icinga\Module\Reporting\Clicommands;
+
+use Exception;
+use Icinga\Module\Pdfexport\ProvidedHook\Pdfexport;
+use Icinga\Module\Reporting\Cli\Command;
+use Icinga\Module\Reporting\Report;
+
+class DownloadCommand extends Command
+{
+    /**
+     * Download report using format types pdf|csv|json (default pdf)
+     *
+     * USAGE:
+     *
+     *   icingacli download <id> --format
+     *
+     */
+    public function indexAction()
+    {
+        $id = $this->params->shiftRequired('id');
+
+        $report = Report::fromDb($id);
+        $format = $this->params->get('format', 'pdf');
+        $name = sprintf(
+            '%s (%s) %s',
+            $report->getName(),
+            $report->getTimeframe()->getName(),
+            date('Y-m-d H:i')
+        );
+
+        switch ($format) {
+            case 'pdf':
+                $content = Pdfexport::first()->htmlToPdf($report->toPdf());
+                break;
+            case 'csv':
+                $content = $report->toCsv();
+                break;
+            case 'json':
+                $content = $report->toJson();
+                break;
+            default:
+                throw new Exception('Format is invalid');
+        }
+
+        $filename = getcwd() . "/$name.$format";
+        file_put_contents($filename, $content);
+        echo "$filename\n";
+        exit;
```

Comment:
The `exit()` call here is not only superfluous, but also incorrect. The CLI command action is executed by a dispatcher, which executes code before and after the action is executed. With `exit()`, the dispatcher cannot execute any code after the action. Please try the download command with the `--benchmark` option with and without the `exit()` call and compare the results.